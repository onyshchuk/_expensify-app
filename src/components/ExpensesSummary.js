import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import getVisibleExpenses from './../selectors/expenses';
import getExpensesTotal from './../selectors/expenses-total';


export const ExpensesSummary = props => {
   const visibleExpenses = getVisibleExpenses(props.expenses, props.filters);
   const expensesTotal = getExpensesTotal(visibleExpenses) / 100;

   return (
      <p>Viewing {visibleExpenses.length} {visibleExpenses.length === 1 ? 'expense' : 'expenses'} totaling {numeral(expensesTotal).format('$0,0.00')}</p>
   );
};

const mapStateToProps = state => ({
   expenses: state.expenses,
   filters: state.filters
});

export default connect(mapStateToProps)(ExpensesSummary);

