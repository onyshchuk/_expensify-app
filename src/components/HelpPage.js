import React from 'react';

const HelpPage = () => (
   <div>
      This is from my help expence component
   </div>
);

export default HelpPage;