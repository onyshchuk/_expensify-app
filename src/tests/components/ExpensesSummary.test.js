import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from './../../components/ExpensesSummary';
import expenses from './../fixtures/expenses';
import { filters } from './../fixtures/filters';

test('should render Expense Summary page properly', () => {
   const wrapper = shallow(<ExpensesSummary
      expenses={expenses}
      filters={filters}
   />);
   expect(wrapper).toMatchSnapshot();
});