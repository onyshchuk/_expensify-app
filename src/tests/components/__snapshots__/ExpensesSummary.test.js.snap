// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expense Summary page properly 1`] = `
<p>
  Viewing 
  3
   
  expenses
   totaling 
  $1,141.95
</p>
`;
